lib/modules/datasource/maven/util.ts

```diff
@@ -17,6 +19,15 @@ import type {
   MavenXml,
 } from './types';
 
+// Singleton S3 instance initialized on-demand.
```

Comment:
s3 stuff should moved to `./s3.ts` file, so it can be mocked and tested more easily.